.pre-commit-config.yaml

```diff
@@ -1,10 +1,59 @@
+exclude: |
+  (?x)(
+    # Python/system files
+    ^.*/__init__\.py$|
+    ^.*?/\.venv/.*$|
+    ^.*?/node_modules/.*$|
+
+    ^.*?/charts/.*$|
+    ^airbyte-integrations/bases/base-normalization/.*$|
+    ^.*?/normalization_test_output/.*$|
+
+    ^.*?/pnpm-lock\.yaml$|
+    ^.*?/source-amplitude/unit_tests/api_data/zipped\.json$|
+
+    # Generated/test files
+    ^airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/.*$|
+    ^.*?/airbyte-ci/connectors/metadata_service/lib/tests/fixtures/.*/invalid/.*$|
+    ^airbyte-ci/connectors/metadata_service/lib/tests/fixtures/.*/invalid/.*$|
+    ^.*?/airbyte-ci/connectors/pipelines/tests/test_format/non_formatted_code/.*$|
+    ^airbyte-ci/connectors/pipelines/tests/test_format/non_formatted_code/.*$|
+    ^.*?/airbyte-integrations/connectors/destination-.*/expected-spec\.json$
+  )
+
 repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.8.3
+    hooks:
+      # Run the linter.
+      - id: ruff
+        args:
+          - --fix
+          - --select=I
+
+      # Run the formatter.
+      - id: ruff-format
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v3.0.3
+    hooks:
+      - id: prettier
+        types_or: [json, yaml]
+        additional_dependencies:
+          - prettier@3.0.3
+
   - repo: local
     hooks:
-      - id: format-fix-all-on-push
-        always_run: true
-        entry: airbyte-ci --disable-update-check format fix all
+      - id: addlicense
+        name: Add license headers
+        entry: addlicense -c "Airbyte, Inc." -l apache -v -f LICENSE_SHORT
+        language: golang
+        additional_dependencies: [github.com/google/addlicense@v1.1.1]
+        files: \.(java|kt|py)$
+
+      - id: spotless
+        name: Format Java files with Spotless
+        entry: bash -c 'command -v mvn >/dev/null 2>&1 || { echo "Maven not installed, skipping spotless" >&2; exit 0; }; mvn -f spotless-maven-pom.xml spotless:apply'
```

Comment:
Did you test running spotless like this?

Maven might generate some garbage, this doesn't matter when running in a dagger container which gets thrown away, but this might not be the case outside of CI from what I can tell.

Furthermore, I'm really not convinced that skipping this formatting step when maven is missing is a good idea. Instead, this should fail hard. 

FYI it's possible to run spotless using gradle, maven only really made sense because of containers. The nice thing about gradle is that it's always present, worst case it will be bootstrapped the first time the `gradlew` script runs.

Reply:
Python folks don't have Spotless on their machines. We should not block them from `pre-commit run --all-files` and send them into 2 hours of computer setup, let alone OSS contributors. 

But to your point, nope, I have not tested this ;)

Reply:
@postmar - fwiw, I do have Maven on my machine and I tested locally without any surprises. I did see that one intentionally ignored andmis-formatted Java files was getting formatted in my test. Hence the update to spotless config here in this PR - which added that exclude so that `pre-commit run -all-files` would be safe to run without generating a diff. 

If you see any issues, I'm happy to jump in to address (pre- or post-merge).

Reply:
> Python folks don't have Spotless on their machines. 

Not sure what your point is here @natikgadzhi. Java folks, which is like half the engineering org, don't have pyenv installed. We shouldn't be hurting their experience either.

> Hence the update to spotless config here in this PR 

Thanks, I'm guessing this is the deliberately-misformatted java file used in an airbyte-ci test.

If the CI is good then I'm good. Local issues I'm happy to help fix as well, in any of Airbyte's programming languages.